disnake/guild.py

```diff
@@ -2224,14 +2232,29 @@ async def create_scheduled_event(
             The request failed.
         TypeError
             The ``image`` asset is a lottie sticker (see :func:`Sticker.read`),
-            or one of ``entity_type``, ``privacy_level``, or ``entity_metadata``
-            is not of the correct type.
+            one of ``entity_type``, ``privacy_level``, or ``entity_metadata``
+            is not of the correct type, or the ``entity_type`` was not provided and
+            could not be assumed from the ``channel``.
 
         Returns
         -------
         :class:`GuildScheduledEvent`
             The newly created guild scheduled event.
         """
+
+        if channel is None:
+            entity_type = GuildScheduledEventEntityType.external
+            channel = MISSING
```

Comment:
```suggestion
        if channel is None or channel is MISSING:
            entity_type = GuildScheduledEventEntityType.external
```
(combined with the other channel change below)

Reply:
IMHO this shouldn't work like this. If channel is unprovided, I think we should require that entity_type was provided rather than default to external.

Reply:
It'll require an additional overload, but in the long run that behavior is better, yeah. I've updated my other comments to reflect this and not default to external if no channel is given.

Reply:
fixed in [fce1fd9](https://github.com/DisnakeDev/disnake/pull/617/commits/fce1fd99f4eb82473fad14e3ce3a417d5fd88e4e)